### PR TITLE
[PREVIEW] - ROC-4198: hands off page

### PR DESCRIPTION
**Pull request by yarpo**
### JIRA link
https://tools.hmcts.net/jira/browse/ROC-4198

### Change description
Add hands off page for directions Questionnaire forms.

To test it locally you need to use this claim store:
https://github.com/hmcts/cmc-claim-store/pull/437

### Work checklist

- [x] Unit tests added where applicable
- [x] Route tests added for new pages
- [x] New pages included in a11y tests
- [x] Task list and task completeness checks updated
- [x] Check and send page updated
- [x] Routes, page content and page flows of new features are toggled 
- [x] UI changes look good on mobile
- [x] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/main/features/dashboard/views/directions-questionnaire.njk [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+{% from "./macro/cmcCorrespondenceAddress.njk" import cmcCorrespondenceAddress %}
+
+{% set heading = 'Complete your directions questionnaire form' %}
+
+{% block content %}
+<div class="grid-row">
+  <div class="column-two-thirds">
+    <p class="panel panel-border-wide">
+      {{ t('We need to receive your response before ' +
+      '<strong class="bold">4pm on <span class="dates5"> {{ deadline }}</span></strong>.',
```
Comment (robertparkinson) [inline src/main/features/dashboard/views/directions-questionnaire.njk]: what is the class dates5? I can't find it any where else

File: src/main/features/dashboard/index.ts [hunk 2]
```diff
@@ -21,17 +22,24 @@ function requestHandler (): express.RequestHandler {
...
-      app.settings.nunjucksEnv.filters.renderPaymentSchedule = (value: string) => {
-        switch (value) {
-          case PaymentSchedule.EACH_WEEK:
-            return 'every week'
-          case PaymentSchedule.EVERY_TWO_WEEKS:
-            return 'every two weeks'
-          case PaymentSchedule.EVERY_MONTH:
-            return 'every month'
+    if (app.settings.nunjucksEnv) {
+
```
Comment (timja) [inline src/main/features/dashboard/index.ts]: remove extra line please

File: src/main/features/dashboard/routes/directions-questionnaire.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export default express.Router()
+  .get(
+    page.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response): Promise<void> => {
+      const { externalId } = req.params
+      const user: User = res.locals.user
+      const claim: Claim = await new ClaimStoreClient().retrieveByExternalId(externalId, user)
+
+      res.render(page.associatedView, {
+        deadline: claim.respondedAt.clone().add(19, 'days'),
```
Comment (timja) [inline src/main/features/dashboard/routes/directions-questionnaire.ts]: you can do a filter in this view for this now ` | addDays(30)`
see:
https://github.com/hmcts/cmc-citizen-frontend/pull/720

Comment (geek-soft) [reply src/main/features/dashboard/routes/directions-questionnaire.ts]: I would advise against doing it in template as this is business logic. Andrew's solution of mutating dates in template is perfectly fine for examples though.

Comment (timja) [reply src/main/features/dashboard/routes/directions-questionnaire.ts]: sure, I think its also, 14 days + 5 days service, if you want to separate it out?

File: src/main/features/dashboard/routes/directions-questionnaire.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const page = Paths.directionsQuestionnairePage
+
+/* tslint:disable:no-default-export */
+export default express.Router()
+  .get(
+    page.uri,
+    ErrorHandling.apply(async (req: express.Request, res: express.Response): Promise<void> => {
+      const { externalId } = req.params
+      const user: User = res.locals.user
+      const claim: Claim = await new ClaimStoreClient().retrieveByExternalId(externalId, user)
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/dashboard/routes/directions-questionnaire.ts]: ok

File: src/main/features/dashboard/views/directions-questionnaire.njk [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      { deadline: deadline | date }) | safe }}
+
+    </p>
+
+    <h3 class="heading-medium">{{ t('Follow these steps') }}</h3>
+
+    <ul class="list list-number">
+      <li>
+        {{ t('Download the <strong class="bold">directions questionnaire form.</strong>') | safe}}
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/dashboard/views/directions-questionnaire.njk]: ok

File: src/main/features/dashboard/views/directions-questionnaire.njk [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      <li>{{ externalLink('Download directions questionnaire form', asset_paths['pdf'] + '/n180-eng.pdf') }} (PDF, 115 KB).
+      </li>
+    </ul>
+
+    <p>{{ t('Use these details when completing the form:') }}</p>
+
+    <ul class="list list-bullet">
+
+      <li>{{ t('In the: County Court Business Centre Online Court') }}</li>
+      <li>{{ t('Claim number: {{ claimNumber }}', {claimNumber: claimNumber }) }}</li>
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/dashboard/views/directions-questionnaire.njk]: ok

File: src/main/features/dashboard/views/macro/cmcCorrespondenceAddress.njk [hunk 7]
```diff
@@ -0,0 +1,9 @@
+{% macro cmcCorrespondenceAddress() %}
+<ul class="panel">
+  <li>{{ t('Civil Money Claims') }}</li>
+  <li>{{ t('St. Katharine’s House') }}</li>
```
[the remark this answers is not in the document]
Comment (yarpo) [reply src/main/features/dashboard/views/macro/cmcCorrespondenceAddress.njk]: Yeah, I was thinking if there is any point of doing it.

I will remove it

File: src/main/features/dashboard/routes/directions-questionnaire.ts (hunk 3 above)
Comment (yarpo) [reply src/main/features/dashboard/routes/directions-questionnaire.ts]: I have left it here as it was. I used `addDays` in template: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk:81
src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk:61

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk [hunk 8]
```diff
@@ -51,21 +51,36 @@
...
       { defendantName: claim.claimData.defendant.name }) }}
   </p>
+  <p>{{ t('{{ defendantName }} has rejected your claim. You’ll have to go to a hearing to resolve it.',
+        { defendantName: claim.claimData.defendant.name }) }}</p>
+
+  <p>{{ t('You need to complete a <a href="{{ url }}">directions questionnaire form</a> to tell us more about the claim.',
+        { url: DashboardPaths.directionsQuestionnairePage.evaluateUri({ externalId: claim.externalId }) }) | safe }}</p>
+
+  <p>{{ t('Your claim will be cancelled if you don’t complete and return the form before 4pm on {{ deadline }}.',
+        { deadline: claim.respondedAt | addDays(19) | date }) }}</p>
```
Comment (geek-soft) [inline src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: As mentioned earlier, please move deadline calculation to route file.

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk [hunk 9]
```diff
@@ -51,21 +51,36 @@
...
-    {{ t('You have rejected the claim. We’ll send you a directions questionnaire – this is a form which you must '
-      + 'complete to tell us more about the claim.') }}
+    {{ t('You’ve rejected the claim and said you don’t want to use mediation to solve it. You’ll have to go to a hearing.') }}
   </p>
+
+  <p>{{ t('You need to complete a <a href="{{ url }}">directions questionnaire form</a> to tell us more about the claim.',
+        { url: DashboardPaths.directionsQuestionnairePage.evaluateUri({ externalId: claim.externalId }) }) | safe }}</p>
+
+  <p>{{ t('Your defence will be cancelled if you don’t complete and return the form before 4pm on {{ deadline }}.',
+        { deadline: claim.respondedAt | addDays(19) | date }) }}</p>
```
Comment (geek-soft) [inline src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: The same here.

Comment (yarpo) [reply src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: done

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk (hunk 8 above)
Comment (yarpo) [reply src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: done

**Comment by geek-soft** [comment]
@robertparkinson, PR ready but we will wait for 'go live' from Clem next week.

**Comment by geek-soft** [comment]
Apparently there is logic change in terms of calculating deadline. More work required by the looks of it.

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk [hunk 10]
```diff
@@ -51,21 +51,36 @@
       + 'this is a form which you must complete to tell us more about the claim.',
       { defendantName: claim.claimData.defendant.name }) }}
   </p>
+  <p>{{ t('{{ defendantName }} has rejected your claim. You’ll have to go to a hearing to resolve it.',
+        { defendantName: claim.claimData.defendant.name }) }}</p>
+
+  <p>{{ t('You need to complete a <a href="{{ url }}">directions questionnaire form</a> to tell us more about the claim.',
```
Comment (jacmurphy) [inline src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: Please use the internalLink macro here

File: src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk [hunk 11]
```diff
@@ -51,21 +51,36 @@
...
 
 {% macro claimRejectedFreeMediationNOForDefendantClaimDetails(claim) %}
   <h2 class="heading-small">{{ t('Your response to the claim') }}</h2>
   <p>
-    {{ t('You have rejected the claim. We’ll send you a directions questionnaire – this is a form which you must '
-      + 'complete to tell us more about the claim.') }}
+    {{ t('You’ve rejected the claim and said you don’t want to use mediation to solve it. You’ll have to go to a hearing.') }}
   </p>
+
+  <p>{{ t('You need to complete a <a href="{{ url }}">directions questionnaire form</a> to tell us more about the claim.',
```
Comment (jacmurphy) [inline src/main/features/dashboard/views/macro/claimStatus/claimRejected.njk]: Please use the internalLink macro here @yarpo

**Pull request merged**